src/api/app/jobs/send_event_emails.rb

```diff
@@ -5,18 +5,17 @@ class SendEventEmails
   attr_accessor :event
 
   def perform
-    Event::Base.where(mails_sent: false).order(:created_at).limit(1000).lock(true).each do |event|
-      event.mails_sent = true
-      begin
-        event.save!
-      rescue ActiveRecord::StaleObjectError
-        # if someone else saved it too, better don't continue - we're not alone
-        return false
-      end
+    Event::Base.where(mails_sent: false).order(created_at: :asc).limit(1000).each do |event|
       subscribers = event.subscribers
-      next if subscribers.empty?
-      EventMailer.event(subscribers, event).deliver_now
+
+      if subscribers.empty?
+        event.update_attributes(mails_sent: true)
+        next
+      end
+
       create_rss_notifications(event)
+      EventMailer.event(subscribers, event).deliver_now
```

Comment:
What we already talked... but at least to have it here.
We should have lines 16 and 17 in a begin-rescue block to make sure that a failing event is marked as sent (like it was before). To not get stuck on events unprocessed over and over again